pkg/apis/core/validation/validation.go

```diff
@@ -2286,6 +2289,41 @@ func validateDNSPolicy(dnsPolicy *core.DNSPolicy, fldPath *field.Path) field.Err
 	return allErrors
 }
 
+const (
+	maxDNSNameservers     = 3
+	maxDNSSearchPaths     = 6
+	maxDNSSearchListChars = 256
+)
+
+func validatePodDNSParams(dnsParams *core.PodDNSParams, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+	if dnsParams != nil {
+		if !utilfeature.DefaultFeatureGate.Enabled(features.CustomPodDNS) {
+			allErrs = append(allErrs, field.Forbidden(fldPath, "DNSParams: custom pod DNS is disabled by feature gate"))
+		} else {
+			if len(dnsParams.Nameservers) > maxDNSNameservers {
+				allErrs = append(allErrs, field.Invalid(fldPath.Child("nameservers"), dnsParams.Nameservers, fmt.Sprintf("must not have more than %v nameservers", maxDNSNameservers)))
+			}
+			for _, ns := range dnsParams.Nameservers {
+				if ip := net.ParseIP(ns); ip == nil {
+					allErrs = append(allErrs, field.Invalid(fldPath.Child("nameservers"), ns, "must be valid IP address"))
+				}
+			}
+			if len(dnsParams.Searches) > maxDNSSearchPaths {
+				allErrs = append(allErrs, field.Invalid(fldPath.Child("searches"), dnsParams.Searches, fmt.Sprintf("must not have more than %v search paths", maxDNSSearchPaths)))
+			}
+			// Include the space between search paths.
+			if len(strings.Join(dnsParams.Searches, " ")) > maxDNSSearchListChars {
+				allErrs = append(allErrs, field.Invalid(fldPath.Child("searches"), dnsParams.Searches, "must not have more than 256 characters (including spaces) in the search list"))
+			}
+			for _, search := range dnsParams.Searches {
+				allErrs = append(allErrs, ValidateDNS1123Subdomain(search, fldPath.Child("searches"))...)
+			}
+		}
+	}
```

Comment:
also validate that all Options have a non-empty name ?

Reply:
Good point, added codes and tests to validate options.